tests/test_requests.py

```diff
@@ -1303,57 +1303,75 @@ def test_response_lines(self):
+        assert list(r.iter_lines(decode_unicode=True, delimiter='\r\n')) == unicode_mock_data.split('\r\n')
+
+        # When delimiter is None, we should yield the same result as splitlines()
+        # which supports the universal newline.
+        # '\r', '\n', and '\r\n' are all treated as one line break.
+
+        # decode_unicode=None, output raw bytes
         result = list(r.iter_lines())
-        assert result != mock_data.splitlines()
-        assert result[2] == ''
-        assert result[4] == ''
+        assert result == mock_data.splitlines()
+
+        # decode_unicode=True, output unicode strings
+        result = list(r.iter_lines(decode_unicode=True))
+        assert result == unicode_mock_data.splitlines()
+
         # If we change all the line breaks to `\r`, we should be okay.
-        mock_chunks = [chunk.replace('\n', '\r') for chunk in mock_chunks]
-        mock_data = ''.join(mock_chunks)
+        # decode_unicode=None, output raw bytes
+        mock_chunks = [chunk.replace(b'\n', b'\r') for chunk in mock_chunks]
+        mock_data = b''.join(mock_chunks)
         assert list(r.iter_lines()) == mock_data.splitlines()
 
+        # decode_unicode=True, output unicode strings
+        unicode_mock_data = mock_data.decode('utf-8')
+        assert list(r.iter_lines(decode_unicode=True)) == unicode_mock_data.splitlines()
+
 
     @pytest.mark.parametrize(
         'content, expected_no_delimiter, expected_delimiter', (
-            ([''], [], []),
-            (['line\n'], ['line'], ['line\n']),
-            (['line', '\n'], ['line'], ['line\n']),
-            (['line\r\n'], ['line'], ['line', '']),
+            ([b''], [], []),
+            ([b'line\n'], ['line'], ['line\n']),
+            ([b'line', b'\n'], ['line'], ['line\n']),
+            ([b'line\r\n'], ['line'], ['line', '']),
             # Empty chunk in the end of stream, same behavior as the previous
-            (['line\r\n', ''], ['line'], ['line', '']),
-            (['line', '\r\n'], ['line'], ['line', '']),
-            (['a\r', '\nb\r'], ['a', '', 'b'], ['a', 'b\r']),
-            (['a\n', '\nb'], ['a', '', 'b'], ['a\n\nb']),
-            (['a\r\n','\rb\n'], ['a', '', 'b'], ['a', '\rb\n']),
-            (['a\nb', 'c'], ['a', 'bc'], ['a\nbc']),
-            (['a\n', '\rb', '\r\nc'], ['a', '', 'b', 'c'], ['a\n\rb', 'c']),
-            (['a\r\nb', '', 'c'], ['a', 'bc'], ['a', 'bc'])  # Empty chunk with pending data
+            ([b'line\r\n', b''], ['line'], ['line', '']),
+            ([b'line', b'\r\n'], ['line'], ['line', '']),
+            ([b'a\r', b'\nb\r'], ['a', 'b'], ['a', 'b\r']),
+            ([b'a\n', b'\nb'], ['a', '', 'b'], ['a\n\nb']),
+            ([b'a\r\n',b'\rb\n'], ['a', '', 'b'], ['a', '\rb\n']),
+            ([b'a\nb', b'c'], ['a', 'bc'], ['a\nbc']),
+            ([b'a\n', b'\rb', b'\r\nc'], ['a', '', 'b', 'c'], ['a\n\rb', 'c']),
+            ([b'a\r\nb', b'', b'c'], ['a', 'bc'], ['a', 'bc'])  # Empty chunk with pending data
```

Comment:
Again, your test seems to show that all the unprefixed strings are intended to be Unicode: they need to be prefixed